RecoTauTag/HLTProducers/src/TauTagFilter.cc

```diff
@@ -0,0 +1,171 @@
+    desc.add<double>("matchingdR", 0.5)->setComment("deltaR for matching with seeds");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& eventsetup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override {
+    if (saveTags())
+      filterproduct.addCollectionTag(tausSrc_);
+
+    int nTauPassed = 0;
+
+    const auto tausHandle = event.getHandle(tausToken_);
+    const auto& taus = *tausHandle;
+
+    const std::vector<LorentzVectorM> seed_p4s = extractMomenta(&event);
+    auto hasMatch = [&](const LorentzVectorM& p4) {
+      for (const auto& seed_p4 : seed_p4s) {
+        if (reco::deltaR2(p4, seed_p4) < matchingdR2_)
+          return true;
+      }
+      return false;
+    };
+
+    const auto& tauTags = event.get(tauTagsToken_);
+    const TauTagCollection* tauPtCorr = nullptr;
+    if (usePtCorr_)
+      tauPtCorr = &event.get(tauPtCorrToken_);
+
+    if (taus.size() != tauTags.size())
+      throw cms::Exception("Inconsistent Data", "PNetTauTagFilter::hltFilter") << "taus.size() != tauTags.size()";
+    if (usePtCorr_ && taus.size() != tauPtCorr->size())
+      throw cms::Exception("Inconsistent Data", "PNetTauTagFilter::hltFilter") << "taus.size() != tauPtCorr.size()";
+
+    for (size_t tau_idx = 0; tau_idx < taus.size(); ++tau_idx) {
+      const auto& tau = taus[tau_idx];
+      double pt = tau.pt();
+      if (usePtCorr_)
+        pt *= (*tauPtCorr)[tau_idx].second;
+      const double eta = std::abs(tau.eta());
+      if (pt > minPt_ && eta < maxEta_ && (!matchWithSeeds_ || hasMatch(tau.polarP4()))) {
+        const double tag = tauTags[tau_idx].second;
+        const double tag_thr = selector_(tau);
+        if (tag > tag_thr) {
+          filterproduct.addObject(trigger::TriggerTau, TauRef(tausHandle, tau_idx));
+          nTauPassed++;
+        }
+      }
+    }
+
+    return nTauPassed >= nExpected_;
+  }
+
+private:
+  std::vector<LorentzVectorM> extractMomenta(const edm::Event* event = nullptr) const {
+    std::vector<LorentzVectorM> seed_p4s;
+    if (matchWithSeeds_) {
+      const trigger::TriggerFilterObjectWithRefs* seeds = nullptr;
+      if (event)
+        seeds = &event->get(seedsSrc_);
+      for (const int seedType : seedTypes_) {
+        if (seedType == trigger::TriggerL1Tau) {
+          extractMomenta<l1t::TauVectorRef>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerL1Jet) {
+          extractMomenta<l1t::JetVectorRef>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerTau) {
+          extractMomenta<std::vector<reco::PFTauRef>>(seeds, seedType, seed_p4s);
+        } else if (seedType == trigger::TriggerJet) {
+          extractMomenta<std::vector<reco::PFJetRef>>(seeds, seedType, seed_p4s);
+        } else
+          throw cms::Exception("Invalid seed type", "PNetTauTagFilter::hltFilter")
```

Comment:
```suggestion
          throw cms::Exception("Invalid seed type", "TauTagFilter::hltFilter")
```
to match the class name.